node/middlewares/checkPublishedApp.ts

```diff
@@ -53,20 +58,66 @@ export async function checkPublishedApp(
 
   const { appId } = buildStatus
   const { name } = parseAppId(appId)
-  let installResponse: InstallResponse = { code: '' }
+  let installResponse: AppInstallResponse = { message: '' }
 
-  try {
-    installResponse = await ctx.clients.billings.installApp(appId, true, false)
-  } catch (err) {
-    logger.error(`Could not install ${name} - ${err}`)
-    await returnResponseError({
-      code: 'INSTALLATION_ERROR',
-      ctx,
-      message: JSON.stringify(installResponse),
-      next,
-    })
+  if (targetWorkspace === ctx.vtex.workspace) {
+    try {
+      installResponse = (await ctx.clients.apps.installApp(
+        appId
+      )) as AppInstallResponse
 
-    return
+      //   ctx.clients.billings.installApp(
+      //   appId,
+      //   true,
+      //   false
+      // )
+    } catch (err) {
+      logger.error(
+        `Could not install ${name} - ${err}, ${JSON.stringify(installResponse)}`
+      )
+      await returnResponseError({
+        code: 'INSTALLATION_ERROR',
+        ctx,
+        message: 'simple installation error',
+        next,
+      })
+
+      return
+    }
+  } else {
+    try {
+      await ctx.clients.workspaces.get(ctx.vtex.account, targetWorkspace)
+    } catch (err) {
+      await ctx.clients.workspaces.create(
+        ctx.vtex.account,
+        targetWorkspace,
+        false
+      )
+    }
+
+    const newCtx = {
+      ...ctx.vtex,
+      timeout: 10000,
+      workspace: targetWorkspace,
+    } as IOContext
+
+    const newApps = new Apps(newCtx)
+
+    try {
+      installResponse = (await newApps.installApp(appId)) as AppInstallResponse
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log('error:', err)
```

Comment:
I think is safe to use console `info` and `error`, can you confirm?
```suggestion
      console.error('error:', err)
```